api/app/c_haines/fetch.py

```diff
@@ -99,22 +91,74 @@ def fetch_network_link_kml() -> str:
     return writer.getvalue()
 
 
-async def fetch_model_runs(model_run_timestamp: datetime):
-    """ Fetch recent model runs """
-    with app.db.database.get_read_session_scope() as session:
-        model_runs = get_model_run_predictions(session, model_run_timestamp)
-
-        result = CHainesModelRuns(model_runs=[])
-        prev_model_run_id = None
-
-        for model_run_id, tmp_model_run_timestamp, name, abbreviation, prediction_timestamp in model_runs:
-            if prev_model_run_id != model_run_id:
-                model_run_predictions = CHainesModelRunPredictions(
-                    model=WeatherPredictionModel(name=name, abbrev=abbreviation),
-                    model_run_timestamp=tmp_model_run_timestamp,
-                    prediction_timestamps=[])
-                result.model_runs.append(model_run_predictions)
-                prev_model_run_id = model_run_id
-            model_run_predictions.prediction_timestamps.append(prediction_timestamp)
+def extract_model_run_prediction_from_path(prediction_path: str) -> Tuple[str, datetime, datetime]:
+    """ Extract model abbreviation, model run timestamp and prediction timestamp from prediction path """
+    name_split = prediction_path.strip('/').split('/')
+
+    prediction_timestamp = datetime.fromisoformat(name_split[-1].rstrip('.json')+'+00:00')
+
+    hour = int(name_split[-2])
```

Comment:
isn't this functionality captured in `extract_model_run_timestamp_from_path()` method immediately below?

Reply:
Ever so slightly different. The on is doing a model run, the other is doing a prediction. I agree they look very similar, but I'd prefer them not to share too much code.